src/Reflection/ClassReflection.php

```diff
@@ -357,9 +405,60 @@ public function getParents(): array
 	 */
 	public function getInterfaces(): array
 	{
-		return array_map(function (\ReflectionClass $interface): ClassReflection {
-			return $this->broker->getClass($interface->getName());
-		}, $this->getNativeReflection()->getInterfaces());
+		$interfaces = [];
+
+		$parent = $this->getParentClass();
+		if ($parent !== false) {
+			foreach ($parent->getInterfaces() as $interface) {
+				$interfaces[$interface->getName()] = $interface;
```

Comment:
What if some interfaces appear multiple times in the hierarchy?

Reply:
`@implements` declarations of children take precedence over `@implements` of the same interface in parents. There should probably be some rule over this, though, because two `@implements` of the same class with differents types might not be allowed.

Reply:
There should be some tests for this behavior if it's not already covered :)